projects/gateway2/extensions2/plugin/plugin.go

```diff
@@ -0,0 +1,103 @@
+package extensionsplug
+
+import (
+	"context"
+
+	envoy_config_cluster_v3 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	envoy_config_endpoint_v3 "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+	"github.com/solo-io/gloo/projects/gateway2/ir"
+	gwv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+	"github.com/solo-io/gloo/projects/gateway2/reports"
+	"istio.io/istio/pkg/kube/krt"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type AttachmentPoints uint
```

Comment:
```suggestion
type AttachmentPoint uint
```

What do you think of singular?

Reply:
this is an bit flags that may have info on more than one point